### PR TITLE
Distinguish resources among different SAVE configs

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Introduced default method `resourceDirectories` in `Plugin` to get resource directories managed by this config
* Added `suiteName` property to `GeneralConfig`

### TODO:
* Call this method in plugin interface

### CODE REVIEW

File: save-common/src/commonMain/kotlin/org/cqfn/save/core/files/FileUtils.kt [hunk 1]
```diff
@@ -72,3 +72,35 @@ fun FileSystem.createFile(path: Path): Path {
...
+ * ```
+ * | directory1
+ * |_ file1
+ * | | directory11
+ * | |_ file2
+ * | |_ directory21
+ * | | directory12
+ * | | directory13
+ * | |_ directory23
+ * | | |_ file33
```
Comment (kgevorkyan) [inline save-common/src/commonMain/kotlin/org/cqfn/save/core/files/FileUtils.kt:88]:
```suggestion
 * |____directory1
 * |       |____file1
 * |       |____directory11
 * |               |____file2
 * |               |____directory21
 * |       |____directory12
 * |       |____directory13
 * |       |____directory23
 * |               |____file33
```

Comment (petertrr) [reply save-common/src/commonMain/kotlin/org/cqfn/save/core/files/FileUtils.kt]: I've changed it to more `tree`-like output, also added a test with the same example.

File: save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt [hunk 2]
```diff
@@ -24,4 +27,17 @@ interface Plugin {
...
+
+    /**
+     * Returns a sequence of directories, where resources for this plugin may be located.
+     * This takes into account, that if underlying directory contains it's own SAVE config,
+     * then this plugin shouldn't touch these resources; it should be done by plugins from that config.
+     *
+     * @param fs a [FileSystem] which is used to traverse the directory hierarchy
+     * @return a sequence of directories possibly containing this plugin's test resources
+     */
+    fun Path.resourceDirectories(fs: FileSystem = FileSystem.SYSTEM): Sequence<Path> = findDescendantDirectoriesBy(true) { file ->
```
[the remark this answers is not in the document]
Comment (petertrr) [reply save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt:39]: I don't really like the idea of a single global object, that we are using to get access to the filesystem (`FileSystem.SYSTEM`). For example, there is `okio-fakefilesystem`, which could be used to run tests with mocked in-memory filesystem, but there is no convenient way to inject it everywhere. I still haven't quite figured out, how to do it better, so I try to at least assign this object to some variables

**Review by kgevorkyan — APPROVED** [review]

File: save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/PluginConfig.kt [hunk 3]
```diff
@@ -11,5 +11,9 @@ interface PluginConfig
 
 /**
  * General configuration for test suite.
+ * @property suiteName name of the test suite
  */
-class GeneralConfig : PluginConfig
+@Suppress("INLINE_CLASS_CAN_BE_USED")
+class GeneralConfig(
+    val suiteName: String,
```
Comment (akuleshov7) [inline save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/PluginConfig.kt:18]: will have conflicts here

**Review by akuleshov7 — APPROVED** [review]
discussed offline, looks good

**Pull request merged**